Formula/nginx-unit.rb

```diff
@@ -0,0 +1,256 @@
+class NginxUnit < Formula
+  desc "Dynamic web and application server for multiple languages"
+  homepage "https://unit.nginx.org/"
+  url "https://unit.nginx.org/download/unit-1.1.tar.gz"
+  sha256 "7c66365f5ea87e7e8903bac9d639676947fe2ab7aa799ce35defc426b3409ee0"
+
+  depends_on "openssl"
+  depends_on "pcre"
+
+  def install
+    openssl = Formula["openssl"]
+    pcre = Formula["pcre"]
+
+    system "./configure",
+      "--prefix=#{prefix}",
+      "--bindir=#{bin}",
+      "--sbindir=#{bin}",
+      # This tells the unit daemon where it can load modules from. Configuring
+      # the location at runtime is painful. We want this to be in the
+      # $HOMEBREW_PREFIX so that other formulae can link in additional modules.
+      "--modules=#{HOMEBREW_PREFIX}/lib/unit",
+      "--state=#{var}/unit",
+      "--log=#{var}/log/unit.log",
+      "--pid=#{var}/run/unit.pid",
+      "--control=unix:#{var}/run/unit.control.sock",
+      "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
+      "--ld-opt=-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    system "make", "install"
+
+    # Beyond here we want to install modules into the keg, so we'll move them
+    # into the prefix ourselves
+    libunit = lib/"unit"
+    libunit.mkpath
+
+    system "./configure", "perl"
+    system "make", "perl"
+    libunit.install "build/perl.unit.so"
+
+    system "./configure", "python"
+    system "make", "python"
+    libunit.install "build/python.unit.so"
+
+    system "./configure", "ruby"
+    system "make", "ruby"
+    libunit.install "build/ruby.unit.so"
+  end
+
+  def post_install
+    (var/"run").mkpath
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using
+      curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+
+      Modules are available for system perl, python and ruby. The system php doesn't
+      support nginx-unit. You can install additional modules into:
+        #{HOMEBREW_PREFIX/"lib/unit"}
+    EOS
+  end
+
+  plist_options :manual => "unitd"
```

Comment:
why not --no-daemon here too?

Reply:
Good point!

Reply:
e.g. https://github.com/Homebrew/homebrew-core/blob/a06f1128a44ed57549bd8ea55d98755491d203c2/Formula/nginx.rb#L146 vs https://github.com/Homebrew/homebrew-core/blob/a06f1128a44ed57549bd8ea55d98755491d203c2/Formula/nginx.rb#L161-L163